bigquery/spec/import_spec.rb

```diff
@@ -0,0 +1,64 @@
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "spec_helper"
+require_relative "../import"
+
+RSpec.describe "Import table data" do
+  before do
+    @sample = Samples::BigQuery::Import.new
+
+    # create a temporary dataset / table
+    require "gcloud"
+    gcloud = Gcloud.new PROJECT_ID
+    @bigquery = gcloud.bigquery
+    @dataset = @bigquery.create_dataset "test_dataset_#{Time.now.to_i}"
+  end
+
+  after do
+    # delete the temporary dataset
+    @dataset.delete force: true
+  end
+
+  [
+    [File.expand_path("spec/data/test_data.json"), "JSON"],
+    [File.expand_path("spec/data/test_data.csv"), "CSV"],
+    ["gs://#{BUCKET_NAME}/test_data.json", "Cloud Storage JSON"],
+    ["gs://#{BUCKET_NAME}/test_data.csv", "Cloud Storage CSV"]
+  ].each do |source, type|
```

Comment:
You can loop to generate the examples but it's not ideal.  Usually we would use Shared Examples for this:
http://rspec.info/documentation/3.5/rspec-core/#Shared_Examples_and_Contexts

As a general rule, though, I try to avoid using shared examples because it usually adds unnecessary complication

Because there isn't lots of redundant code, I wouldn't DRY this too much ... I'd do something explicit like:

```ruby
require_relative "spec_helper"
require_relative "../import"
require "gcloud"

RSpec.describe "Import table data" do

  before do
    @project_id  = ENV["GCLOUD_PROJECT"]
    @bucket_name = ENV["BUCKET"]
    @gcloud      = Gcloud.new ENV["GCLOUD_PROJECT"]
    @bigquery    = @gcloud.bigquery
    @dataset     = @bigquery.create_dataset "test_dataset_#{Time.now.to_f}"
    @dataset.create_table "test_table" do |schema|
      schema.string "name", mode: :required
      schema.string "title", mode: :required
    end
  end

  after do
    @dataset.delete force: true
  end

  it "import JSON" do
    source = File.expand_path "data/test_data.json", __dir__

    expect { import @project_id, @dataset.dataset_id, "test_table", source }.
      to output(/Data imported successfully/).to_stdout

    results = @bigquery.query "SELECT * FROM #{@dataset.dataset_id}.test_table"

    expect(results.size).to eq 3
    expect(result[0]).to include("name" => "Alice")
    expect(result[1]).to include("name" => "Bob")
  end

  it "import CSV" do
    source = File.expand_path "data/test_data.csv", __dir__

    expect { import @project_id, @dataset.dataset_id, "test_table", source }.
      to output(/Data imported successfully/).to_stdout

    results = @bigquery.query "SELECT * FROM #{@dataset.dataset_id}.test_table"

    expect(results.size).to eq 3
    expect(result[0]).to include("name" => "Alice")
    expect(result[1]).to include("name" => "Bob")
  end

  it "import JSON from Cloud Storage" do
    source = "gs://#{@bucket_name}/test_data.json"

    expect { import @project_id, @dataset.dataset_id, "test_table", source }.
      to output(/Data imported successfully/).to_stdout

    results = @bigquery.query "SELECT * FROM #{@dataset.dataset_id}.test_table"

    expect(results.size).to eq 3
    expect(result[0]).to include("name" => "Alice")
    expect(result[1]).to include("name" => "Bob")
  end

  it "import CSV from Cloud Storage" do
    source = "gs://#{@bucket_name}/test_data.csv"

    expect { import @project_id, @dataset.dataset_id, "test_table", source }.
      to output(/Data imported successfully/).to_stdout

    results = @bigquery.query "SELECT * FROM #{@dataset.dataset_id}.test_table"

    expect(results.size).to eq 3
    expect(result[0]).to include("name" => "Alice")
    expect(result[1]).to include("name" => "Bob")
  end
end
```